### ISSUE TITLE
Dealing with conditional hyperparameters

### ISSUE THREAD

**Issue by jredondopizarro** [SheffieldML/GPyOpt#241]
Greetings.

I would like to set up an experiment with  `number_of_layers` and `number_of_neurons_per_each_layer` hyperparameters, so I would need, for example, to exclude from the search space  the `number_of_neurons_in layer_4th` if the `number_of_layers` is < 4.
Is there some way of doing this?

Thanks in advance. #

**Comment by apaleyes** [comment]
I don't think so. Although I can imagine using context and setting parameters you'd like to exclude to some fixed values, say 0. Not quite the same though, but seems close in your case. Check out tutorial on context here: https://nbviewer.jupyter.org/github/SheffieldML/GPyOpt/blob/master/manual/GPyOpt_context.ipynb

**Comment by javiergonzalezh** [comment]
I don' think that you can handle that with context, you need either an [special kernel](https://ml.informatik.uni-freiburg.de/papers/13-BayesOpt_Arc-Kernel.pdf) for that or some from of [tree structure ](http://proceedings.mlr.press/v70/jenatton17a.html) to handle multiple GPs. I have some code for both that is hard to integrate here at this point, because they correspond to an old version of GPyOpt. However, given the progress in [Emukit ](https://github.com/amzn/emukit) and that most development is happening there at the moment we should consider integrate these methods there. The use case is obvious as @jredondopizarro is describing.
